src/invidious/videos/parser.cr

```diff
@@ -456,11 +462,12 @@ def parse_video_info(video_id : String, player_response : Hash(String, JSON::Any
     # Music section
     "music" => JSON.parse(music_list.to_json),
     # Author infos
-    "author"          => JSON::Any.new(author || ""),
-    "ucid"            => JSON::Any.new(ucid || ""),
-    "authorThumbnail" => JSON::Any.new(author_thumbnail.try &.as_s || ""),
-    "authorVerified"  => JSON::Any.new(author_verified || false),
-    "subCountText"    => JSON::Any.new(subs_text || "-"),
+    "author"             => JSON::Any.new(author || ""),
+    "ucid"               => JSON::Any.new(ucid || ""),
+    "authorThumbnail"    => JSON::Any.new(author_thumbnail.try &.as_s || ""),
+    "authorVerified"     => JSON::Any.new(author_verified || false),
+    "subCountText"       => JSON::Any.new(subs_text || "-"),
+    "invidiousCompanion" => JSON::Any.new(subs_text),
```

Comment:
I forgot to document a bit of this. Basically this is an additional section for invidious companion for storing extra useful info like:
- which invidious companion was used to retrieve the video streams (don't forget I allow using multiple ones)

It's sent in https://github.com/iv-org/invidious-companion/blob/e4140a9e1e5807e3890164fb199accb56fec5ed0/src/lib/helpers/youtubePlayerHandling.ts#L92-L94

For the moment it's only that, but I have some ideas like storing which proxy was used to retrieve the video streams.

It's useful to store it because it's cached in the postgresql database and can be used later for knowing like which invidious companion was used to retrieve the video streams.

Reply:
Instead it's a mistake, I didn't put too much attention to `subs_text`, sorry.

Don't you need it to parse it in `extract_video_info()` for then storing it in `params = {`: https://github.com/unixfox/invidious/blob/2683b24eed477b311025f3dfdd44625d99b05ecc/src/invidious/videos/parser.cr#L436

Reply:
`parse_video_info()` extracts the metadata from the player/next endpoint and is returs that `params` Hash to the caller (`extract_video_info()`) here:
https://github.com/iv-org/invidious/blob/2683b24eed477b311025f3dfdd44625d99b05ecc/src/invidious/videos/parser.cr#L104

After that, there is the various bypass logic, to get working streams, then multiple other fields are copied over from the player response. None of these were already in `params` prior to that.
https://github.com/iv-org/invidious/blob/2683b24eed477b311025f3dfdd44625d99b05ecc/src/invidious/videos/parser.cr#L147-L160